src/osd/OSD.cc

```diff
@@ -8725,6 +8751,34 @@ void OSD::enqueue_op(PG *pg, OpRequestRef& op)
 	   << " cost " << op->get_req()->get_cost()
 	   << " latency " << latency
 	   << " " << *(op->get_req()) << dendl;
+
+  // 15 --> op lock completion
+  if (op->get_req()->get_type() == MSG_OSD_REPOPREPLY) {
+    MOSDRepOpReply *r = static_cast<MOSDRepOpReply *>(op->get_req());
+    r->finish_decode();
+    spg_t pgid = pg->get_pgid();
+    const pg_pool_t* pp = osdmap->get_pg_pool(pgid.pool());
+    if (r->get_result() == MSG_OSD_OP_COMP_RESULT && 
+	pp->get_type() == pg_pool_t::TYPE_REPLICATED) {
+      
+
+      PGBackend * pg_backend = pg->get_pgbackend();
+      if (!pg_backend) {
+	dout(0) << __func__ << " PGBackend is null " << dendl;
+	pg->unlock();
```

Comment:
Yes, I intend to avoid existing I/O path with PG lock because the role of ack is simply notifying completion of op. also, sending ack is very short path and enqueue to current sharded work queue cause contention. Therefore, I think ack can be issued without PG lock and enqueuing via op lock completion path. (updating PG related data will be processed at lazy_completion_func or comp_thread.)